db/music.dbml

```diff
@@ -0,0 +1,299 @@
+Table submission {
+  submission_id uuid [pk, default: `gen_random_uuid()`]
+  participant_id int [not null, ref: > participant.participant_id]
+  bracket_id int [not null, ref: > bracket.bracket_id]
+  song_id int [not null, ref: > song.song_id]
+  order int [not null]
+  date_submitted timestamp [not null, default:`now()`]
+  active bool_int [not null]
+  is_dupe bool_int [not null]
+  title text [not null]
+  version text [null]
+  artist text [null]
+  collaborators text [null]
+  composer text [null]
+  link text [not null]
+  genre text [null]
+  emoji text [null]
+  comments text [null]
+
+  indexes {
+    (participant_id, bracket_id, song_id) [unique]
+  }
+}
+
+Table submission_history {
+  submission_history_id uuid [pk, default: `gen_random_uuid()`]
+  date_update timestamp [not null]
+  note_update text [not null, Note: 'dupe ineligible withdrawn "play-in loser"' ]
+  submission_id int [not null, ref: > submission.submission_id]
+  participant_id int [not null, ref: > participant.participant_id]
+  bracket_id int [not null, ref: > bracket.bracket_id]
+  song_id int [not null, ref: > song.song_id]
+  order int [not null]
+  date_submitted timestamp [not null, default:`now()`]
+  active bool_int [not null]
+  is_dupe bool_int [not null]
+  title text [not null]
+  version text [null]
+  artist text [null]
+  collaborators text [null]
+  composer text [null]
+  link text [not null]
+  genre text [null]
+  emoji text [null]
+  comments text [null]
+
+  indexes {
+    submission_id
+    participant_id
+    bracket_id
+    song_id
+  }
+
+  Note: 'composite unique index on submission_id, date_function (https://stackoverflow.com/a/26304364/2506158)'
+}
+
+Table link_relation {
+  link_realtion_id uuid [pk, default: `gen_random_uuid()`]
+  relation_id int [not null]
+  link_id int [not null, ref: > link.link_id]
+  type_link text [not null, Note: 'submission song seed' ]
+
+  indexes {
+    relation_id
+  }
+}
+
+Table link {
+  link_id uuid [pk, default: `gen_random_uuid()`]
+  url text [not null]
+  region text[] [null, Note: 'US Canada Japan Australia Europe Asia "North America" "South America" Africa Worldwide' ]
+}
+
+Table song {
+  song_id uuid [pk, default: `gen_random_uuid()`]
+  title text [not null]
+  title_english text [null]
+  version text [null]
+  artist text [not null, Note: 'CALCULATED']
+  collaborators text [null, Note: 'CALCULATED']
+  composer text [null]
+  year_min int [null]
+  year_max int [null]
+
+  indexes {
+    (title, version, artist, collaborators) [unique]
```

Comment:
It might not hurt to have `title` indexed for full text search (GIN or GiST apparently), though we might want to duplicate the data into a specially formatted column, removing special characters or something maybe. But I wouldn't index on the other things, and I definitely wouldn't make this unique. With our luck we'll get two completely different bands with the same name each releasing a song by the same name in the same year.

Reply:
Hah. That seems unlikely. I guess with that number of fields and the inevitable variations, the unique bit won't do us much good.